restcomm/restcomm.sms/src/main/java/org/restcomm/connect/sms/smpp/SmppClientOpsThread.java

```diff
@@ -411,7 +447,34 @@ public PduResponse firePduRequestReceived(PduRequest pduRequest) {
                     } else {
                        if(logger.isInfoEnabled()) {
                            logger.info("Message is not normal request, esmClass:" + esmClass +", Using message from message body " + decodedPduMessage);
-                        }
+                       }
+
+                       //TODO: point to an array of provider specific functions
+                       Map<String,String> dlr = getMessageIdNexmo(decodedPduMessage);
+                       String messageId = dlr.get("id");
+                       String status = dlr.get("status");
+                       DateTime dateSent = null;
+                       try {
+                           //TODO: abstract Nexmo specific format
+                           dateSent = DateTime.parse(dlr.get("sent_date"), DateTimeFormat.forPattern("yyMMddkkmm"));
```

Comment:
can we save the pattern to be reused?, I guess compiling the pattern is resource consuming...